kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/balancer/LoadBalancer.java

```diff
@@ -97,8 +136,61 @@ public List<String> getFailedEndpoints() {
     /*
      *  background operations for checking failed endpoints
      */
-    public Future<?> checkFailedEndpoints() {
-        return executorService.submit(new CheckFailedEndpoints());
+    public synchronized Future<?> checkFailedEndpoints() {
+        if (!checkFailedEndpoint) {
+            return null;
+        }
+
+        CountDownLatch latch = new CountDownLatch (1);
+        EndpointListener lst = new EndpointListener() {
+            @Override
+            public void markAsActive(String endpoint) {
+                synchronized (endpointListeners) {
+                    endpointListeners.remove(this);
+                }
+                latch.countDown();
+            }
+        };
+        endpointListeners.add(lst);
+        // create a bridge between the latch and the latch
+        Future<Object> futureBridge = new Future<Object> () {
+
+            @Override
+            public boolean cancel(boolean mayInterruptIfRunning) {
+                return false;
+            }
+
+            @Override
+            public boolean isCancelled() {
+                return false;
+            }
+
+            @Override
+            public boolean isDone() {
+                return latch.getCount() == 0;
+            }
+
+            @Override
+            public Object get() throws InterruptedException, ExecutionException {
+                latch.await();
+                return null;
+            }
+
+            @Override
+            public Object get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+                latch.await(timeout, unit);
```

Comment:
Spotbug issue:

`[ERROR] Return value of java.util.concurrent.CountDownLatch.await(long, TimeUnit) ignored in org.kie.server.client.balancer.LoadBalancer$2.get(long, TimeUnit) [org.kie.server.client.balancer.LoadBalancer$2] At LoadBalancer.java:[line 181] RV_RETURN_VALUE_IGNORED`

Reply:
not bug at all. it does not return any value. just waits.